packages/coil-oauth-scripts/src/Stream.ts

```diff
@@ -316,7 +268,62 @@ export class Stream {
     this.monetization.dispatchMonetizationProgressEvent(detail)
   }
 
-  pause() {
+  private setState({
+    state,
+    finalized
+  }: {
+    state: MonetizationState
+    finalized?: boolean
+  }) {
+    this.state = state
+    this.monetization.setState({ state, finalized })
+  }
+
+  async pause() {
     return this.stop(false)
   }
 }
+
+function btpTokenThroughput(token: string): number {
+  const payload = Buffer.from(token.split('.')[1], 'base64')
+  const { throughput } = JSON.parse(payload.toString())
+  if (throughput === undefined) throw new Error('throughput not found')
+  return throughput
+}
+
+async function firstMinuteBandwidth(
+  stream: IlpStream.DataAndMoneyStream,
```

Comment:
Two things to note:
1. If the `stream` closes early for some reason (expired token reconnect?) this first minute behavior is skipped 
2. If it's started with a partially used token, won't that mean it won't get a full minutes worth?

Reply:
I'm not entirely sure the reason for it, but I've been inspecting the amounts given and more often than not, I don't see the final 30 seconds worth getting paid: https://gist.github.com/sublimator/88ecf832ee4454577108ecc27f1a5d0a#file-gistfile1-txt-L76

Reply:
This also means the first minute logic is interruptible. 
Open logger page, wait first payment event, flip to another tab, then back:
https://gist.github.com/sublimator/d62010f79c153e663a1018de568a329e#file-gistfile1-txt-L10

Reply:
And actually, it seems if you do that, it won't pay for minutes:
https://gist.github.com/sublimator/9bef500e6b3335c83ae25cff21deb372

Reply:
> If the stream closes early for some reason (expired token reconnect?) this first minute behavior is skipped
> If it's started with a partially used token, won't that mean it won't get a full minutes worth?
> This also means the first minute logic is interruptible.

Correct. In each of those cases, it just means it falls back to post-pay a little early, which is fine.

> And actually, it seems if you do that, it won't pay for minutes
That's a bug; I'll push a fix.

Reply:
> it just means it falls back to post-pay a little early, which is fine

If it's "fine" do we even need this beyond the first payment ?